model/project_ref.go

```diff
@@ -1255,6 +1304,180 @@ func (projectRef *ProjectRef) Upsert() error {
 	return err
 }
 
+// DefaultSectionToRepo modifies a subset of the project ref to use the repo values instead.
+// This subset is based on the pages used in Spruce.
+func (p *ProjectRef) DefaultSectionToRepo(section ProjectRefSection, caller string) error {
+	before, err := GetProjectSettingsEvents(p)
+	if err != nil {
+		return errors.Wrap(err, "error getting before project settings event")
+	}
+
+	switch section {
+	// check if the sections cover all the settings
+	case ProjectRefGeneralSection:
+		err = db.Update(ProjectRefCollection,
+			bson.M{ProjectRefIdKey: p.Id},
+			bson.M{
+				"$unset": bson.M{
+					ProjectRefEnabledKey:                 1,
+					ProjectRefBatchTimeKey:               1,
+					ProjectRefRemotePathKey:              1,
+					projectRefSpawnHostScriptPathKey:     1,
+					projectRefDispatchingDisabledKey:     1,
+					ProjectRefDeactivatePreviousKey:      1,
+					projectRefRepotrackerDisabledKey:     1,
+					projectRefDefaultLoggerKey:           1,
+					projectRefCedarTestResultsEnabledKey: 1,
+					projectRefPatchingDisabledKey:        1,
+					projectRefTaskSyncKey:                1, // TODO: look into whether this works for individual settings
+					ProjectRefDisabledStatsCacheKey:      1,
+					ProjectRefFilesIgnoredFromCacheKey:   1,
+				},
+			})
+		if err != nil {
+			return errors.Wrapf(err, "error defaulting to repo for section '%s'", section)
+		}
+	case ProjectRefAccessSection:
+		err = db.Update(ProjectRefCollection,
+			bson.M{ProjectRefIdKey: p.Id},
+			bson.M{
+				"$unset": bson.M{
+					ProjectRefPrivateKey:    1,
+					ProjectRefRestrictedKey: 1,
+					ProjectRefAdminsKey:     1,
+				},
+			})
+		if err != nil {
+			return errors.Wrapf(err, "error defaulting to repo for section '%s'", section)
+		}
+	case ProjectRefVariablesSection:
+		err = db.Update(ProjectVarsCollection,
+			bson.M{ProjectRefIdKey: p.Id},
+			bson.M{
+				"$unset": bson.M{
+					projectVarsMapKey:    1,
+					privateVarsMapKey:    1,
+					restrictedVarsMapKey: 1,
+				},
+			})
+		if err != nil {
+			return errors.Wrapf(err, "error defaulting to repo for section '%s'", section)
+		}
+	case ProjectRefGithubSection:
+		err = db.Update(ProjectRefCollection,
+			bson.M{ProjectRefIdKey: p.Id},
+			bson.M{
+				"$unset": bson.M{
+					projectRefPRTestingEnabledKey:      1,
+					projectRefGithubChecksEnabledKey:   1,
+					projectRefGithubTriggerAliasesKey:  1,
+					projectRefGitTagVersionsEnabledKey: 1,
+					ProjectRefGitTagAuthorizedUsersKey: 1,
+					ProjectRefGitTagAuthorizedTeamsKey: 1,
+					projectRefCommitQueueKey:           1,
```

Comment:
The projectRefCommitQueueKey includes Enabled, MergeMethod, Message. 

The UI commit queue section also has the following fields that probably need to be defaulted as well:
- patch type
- commit queue patch definitions.

Reply:
Patch Type is actually outdated; commit queues now accept both PR and CLI. I don't think we need to ask Lara to modify the design but we should drop a comment that it's not relevant anymore.

Commit queue patch definitions are handled in the alias section below; these are aliases with the __commit_queue alias label.

Reply:
Do we not want to do it as part of case ProjectRefGithubAndCQSection: so that when someone clicks on the "default page to repo" button for the commit queue page that gets defaulted as well?